File: app/src/main/java/com/tyoma/testingzone/libs/callback/MyFTPTransferCallback.java
package com.tyoma.testingzone.libs.callback;

// callback of data transfer
public interface MyFTPTransferCallback {
    int START = 1;
    int TRANSFER = 2;
    int COMPLETED = 3;
    int ERROR = 4;
    int ABORTED = 5;

    void onStateChanged(int state);

    void onTransferDone(long fileSize, int resultSize);

    void onErr(int code, String msg);
}

File: app/src/main/java/com/tyoma/testingzone/libs/callback/MyFTPTransferCallback.kt
package com.tyoma.testingzone.libs.callback

// callback of data transfer
interface MyFTPTransferCallback {
    fun onStateChanged(state: Int)

    fun onTransferDone(fileSize: Long, resultSize: Int)

    fun onErr(code: Int, msg: String)

    companion object {
        const val START: Int = 1
        const val TRANSFER: Int = 2
        const val COMPLETED: Int = 3
        const val ERROR: Int = 4
        const val ABORTED: Int = 5
    }
}
